Makefile.dune

```diff
@@ -1,122 +1,109 @@
 	@echo "  - refman-html: build Coq's reference manual [HTML version]"
 	@echo "  - refman-pdf:  build Coq's reference manual [PDF version]"
 	@echo "  - stdlib-html: build Coq's Stdlib documentation [HTML version]"
 	@echo "  - apidoc:      build ML API documentation"
-	@echo "  - release:     build Coq in release mode"
+	@echo ""
+	@echo " Miscellaneous targets:"
 	@echo ""
 	@echo "  - fmt:     run ocamlformat on the codebase"
 	@echo "  - ocheck:  build for all supported OCaml versions [requires OPAM]"
 	@echo "  - ireport: build with optimized flambda settings and emit an inline report"
 	@echo "  - clean:   remove build directory and autogenerated files"
 	@echo "  - help:    show this message"
+	@echo ""
+	@echo " To run an app \\in {coqc,coqtop,coqbyte,coqide}:"
+	@echo ""
+	@echo "  - use 'dune exec -- dev/shim/app-prelude args'"
+	@echo ""
+	@echo " Provided opam/dune packages are:"
+	@echo ""
+	@echo "  - coq: base Coq package, toplevel compilers, tools, stdlib, no GTK"
+	@echo "  - coqide-server: XML protocol language server"
+	@echo "  - coqide: CoqIDE gtk application"
+	@echo ""
+	@echo " To build a package, you can use:"
+	@echo ""
+	@echo "  - 'dune build package.install' : build package in developer mode"
+	@echo "  - 'dune build -p package' : build package in release mode"
+	@echo ""
+	@echo " Packages _must_ be installed using release mode, use: 'dune install -p package'"
+	@echo " See Dune documentation for more information."
 
-# We need to bootstrap with a dummy coq.plugins.ltac so install targets do work.
-plugins/ltac/dune:
-	@echo "(library (name ltac_plugin) (public_name coq.plugins.ltac) (modules_without_implementation extraargs extratactics))" > plugins/ltac/dune
-
-voboot: plugins/ltac/dune
-	dune build --build-dir=$(BOOT_DIR) $(DUNEOPT) @vodeps
-	dune exec --build-dir=$(BOOT_DIR) -- ./tools/coq_dune.exe $(BOOT_CONTEXT)/.vfiles.d
+voboot:
+	@echo "This target is empty and not needed anymore"
 
-states: voboot
+states:
 	dune build --display=short $(DUNEOPT) dev/shim/coqtop-prelude
 
 NONDOC_INSTALL_TARGETS:=coq.install coqide-server.install coqide.install
 
-world: voboot
+world:
 	dune build $(DUNEOPT) $(NONDOC_INSTALL_TARGETS)
 
-coq: voboot
-	dune build $(DUNEOPT) coq.install
-
-coqide: voboot
-	dune build $(DUNEOPT) coqide.install
-
-coqide-server: voboot
-	dune build $(DUNEOPT) coqide-server.install
-
-watch: voboot
+watch:
 	dune build $(DUNEOPT) $(NONDOC_INSTALL_TARGETS) -w
 
-check: voboot
+check:
 	dune build $(DUNEOPT) @check
 
-COQTOP_FILES=ide/idetop.bc ide/coqide_main.bc checker/coqchk.bc
-PLUGIN_FILES=$(wildcard plugins/*/*.mlpack)
-PRINTER_FILES=dev/top_printers.cma
-QUICKBYTE_TARGETS=$(COQTOP_FILES) $(PLUGIN_FILES:.mlpack=.cma) $(PRINTER_FILES) topbin/coqtop_byte_bin.bc
-QUICKOPT_TARGETS=$(COQTOP_FILES:.bc=.exe) $(PLUGIN_FILES:.mlpack=.cmxs) $(PRINTER_FILES:.cma=.cmxa) topbin/coqtop_bin.exe
-
-quickbyte: voboot
-	dune build $(DUNEOPT) $(QUICKBYTE_TARGETS)
-
-quickopt: voboot
```

Comment:
I frequently use `make -f Makefile.dune quickopt` to make sure that I haven't broken any ML code. What should I use instead? `dune build @check`?

Reply:
Indeed, if that's your goal, `dune build @check` is more complete and much faster as it uses the non-optimizing compiler.

Reply:
My view is that these targets have been replaced by the shims, they do indeed depend on exactly what's needed for the desired use case.